dexlib2/src/main/java/org/jf/dexlib2/writer/DexWriter.java

```diff
@@ -221,6 +221,16 @@ public int compare(Entry<? extends CallSiteKey, Integer> o1, Entry<? extends Cal
         };
     }
 
+    private static <T extends Comparable<? super T>> Comparator<Map.Entry<? extends T, ?>> comparableValueComparator() {
+        return new Comparator<Entry<? extends T, ?>>() {
+            @Override public int compare(Entry<? extends T, ?> o1, Entry<? extends T, ?> o2) {
+                int o1_int = Integer.parseInt(o1.getValue().toString());
+                int o2_int = Integer.parseInt(o2.getValue().toString());
+                return Integer.compare(o1_int, o2_int);
+            }
+        };
+    }
+
```

Comment:
why the trip to string? are the values not all Integer? then create a `comparableIntegerValueComparator` and type it accordingly.

better yet, make `comparableValueComparator` really generic. what is wrong with this?
```
<T extends Comparable<? super T>> Comparator<Map.Entry<?, ? extends T>> comparableValueComparator() {
        return new Comparator<Entry<?, ? extends T>>() {
            @Override public int compare(Entry<?, ? extends T> o1, Entry<?, ? extends T> o2) {
                return o1.getValue().compareTo(o2.getValue());
            }
        }
    }
```

Reply:
Agreed. I like this better.